File: src/p0011/java/src/Solution.java
class Solution {
    public int maxArea(int[] h) {
        int result = 0, i = 0, j = h.length - 1;
        while (i < j) {
            if (h[i] < h[j]) {
                result = Math.max(result, (j - i) * h[i]);
                i++;
            } else {
                result = Math.max(result, (j - i) * h[j]);
                j--;
            }
        }
        return result;
    }
}

File: src/p0011/cpp/solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;


class Solution {
public:
    int maxArea(const vector<int> &hs) {
        int result = 0, i = 0, j = hs.size() - 1;
        while (i < j) {
            if (hs[i] < hs[j]) {
                result = max(result, (j - i) * hs[i]);
                i++;
            } else {
                result = max(result, (j - i) * hs[j]);
                j--;
            }
        }
        return result;
    }
};
